pets/peer-finder/peer-finder.go

```diff
@@ -74,17 +75,42 @@ func main() {
 	if ns == "" {
 		ns = os.Getenv("POD_NAMESPACE")
 	}
-	if *svc == "" || ns == "" || (*onChange == "" && *onStart == "") {
-		log.Fatalf("Incomplete args, require -on-change and/or -on-start, -service and -ns or an env var for POD_NAMESPACE.")
-	}
-
 	hostname, err := os.Hostname()
 	if err != nil {
 		log.Fatalf("Failed to get hostname: %s", err)
 	}
+	var domainName string
+
+	// If domain is not provided, try to get it from resolv.conf
+	if (*domain == "") || (ns == "") {
+		resolvConfBytes, err := ioutil.ReadFile("/etc/resolv.conf")
+		resolvConf := string(resolvConfBytes)
+		if err != nil {
+			log.Fatal("Unable to read /etc/resolv.conf")
+		}
+		if ns != "" {
+			tempResolv := strings.Split(resolvConf, ("search " + ns + "."))
```

Comment:
> I think we can support the current options and have it deduce the "default" domain name if none of these variables are supplied

This makes sense to me. Based on that conformance test, it seems safe enough to try and deduce the domain from the `resolv.conf.` I'll let @kubernetes/sig-network-misc take a quick look just in case.

OTOH, I think we can refactor this bit of the code to rely less on string parsing and perhaps use regular expressions instead.